### PR TITLE
Show branch and commit instead of tag when not on release branch

### LINKED ISSUE #627: Building main branch reports v3.0.0 in `gaiad version`
## Steps to Reproduce

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by shahankhatch**
Closes: #627 

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/main/README.md#merging-a-pr))

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/make -f
 
-PACKAGES_SIMTEST=$(shell go list ./... | grep '/simulation')
-VERSION := $(shell echo $(shell git describe --tags) | sed 's/^v//')
```
Comment (alessio) [inline Makefile:4]: `git describe`'s output does show commit information. What's wrong with that?

Comment (alessio) [reply Makefile:4]: ref https://git-scm.com/docs/git-describe

**Review by zmanian — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (shahankhatch) [reply Makefile:4]: I'm thinking of using the branch name instead of the tag since it avoids picking up an old tag (v3.0.0) on the main branch.

Comment (alessio) [reply Makefile:4]: Oh that's because we don't have annotated tags in `main`. Ok makes sense :+1:

**Review by alessio — APPROVED** [review]

**Pull request merged**